jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/URIUtil.java

```diff
@@ -1642,20 +1677,45 @@ else if (!uriA.getAuthority().equals(uriB.getAuthority()))
     }
 
     /**
+     * Add a sub path to an existing URI.
+     *
      * @param uri A URI to add the path to
      * @param path A decoded path element
+     * @param encodePath true to encode provided path, false to leave it alone in resulting URI
      * @return URI with path added.
+     * @see #addPaths(String, String)
      */
-    public static URI addPath(URI uri, String path)
+    public static URI addPath(URI uri, String path, boolean encodePath)
```

Comment:
I'd prefer 2 methods rather than one with a boolean arg

Reply:
I'll just get rid of the boolean.
There's nothing in mainline code right now using it as true anyway.